context_test.go

```diff
@@ -0,0 +1,35 @@
+package clickhouse
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestContext(t *testing.T) {
+	t.Run("call context multiple times making sure query options are persisted across calls",
+		func(t *testing.T) {
+			ctx := Context(context.Background(), WithQueryID("a"))
+			ctx = Context(ctx, WithQuotaKey("b"))
+			ctx = Context(ctx, WithSettings(Settings{
+				"c": "d",
+			}))
+
+			opts := queryOptions(ctx)
+			require.Equal(t, "a", opts.queryID)
+			require.Equal(t, "b", opts.quotaKey)
+			require.Equal(t, "d", opts.settings["c"])
```

Comment:
`require` is more powerful than `assert` in testify lib. It not only does an assertion but also fails immediately. 

It's more helpful if you want to bootstrap some test dependencies and fail fast. For pure assertions, I recommend switching to `assert` package as it gives more transparency on all failing assertions.

Reply:
Switched over to asserts.